crates/noirc_evaluator/src/ssa/optimizations.rs

```diff
@@ -327,9 +327,14 @@ fn cse_block_with_anchor(
                     new_list.push(*ins_id);
                 }
                 Operation::Return(..) => new_list.push(*ins_id),
-                Operation::Intrinsic(_, args) => {
+                Operation::Intrinsic(opcode, args) => {
                     //Add dummy load for function arguments and enable CSE only if no array in argument
                     let mut activate_cse = true;
+                    // We do not want to replace any print intrinsics as we want them to remain in order and unchanged
+                    if let builtin::Opcode::Println(_) = opcode {
+                        activate_cse = false
```

Comment:
Is this necessary? I did not think cse did any reordering. As long as the println call duplicates are not removed we should be fine, right?

Reply:
We should remove the leading zeroes from the output as well. I think we should also prepend a `0x` though to signal it is in hex.

Reply:
I will add the prepend. It will be clearer for users and lets the copy paste output from `println` as inputs to a circuit if need be.

Reply:
I removed the leading zeroes from the output as well.
This is the output of the `strings` test now:
```
0xa
0x32
0xa
[0x1, 0x2, 0x3, 0x5, 0x8]
helld world
[1ed38f2b9f2acc42cc9a62d2202c667582eb5108dc13304651fbf3ae7ce6c6b2,2947812ac2010e25b8f606adf3260a7d2aacba02566b279dcc8d77bd7df1b62b]
```

Reply:
I changed this once more to make sure that we have an even number of digits in the hex string. If a user were to try and use this println output as a hex string in a circuit or somewhere else it would fail to parse. I just check whether the trimmed hex string has an odd number of digits. If this is the case, I simply prepend a zero. The output now looks like this:
```
0x0a
0x32
0x0a
[0x01, 0x02, 0x03, 0x05, 0x08]
helld world
[1ed38f2b9f2acc42cc9a62d2202c667582eb5108dc13304651fbf3ae7ce6c6b2,2947812ac2010e25b8f606adf3260a7d2aacba02566b279dcc8d77bd7df1b62b]
```